src/foam/util/Password.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.util;
+
+import org.bouncycastle.crypto.PBEParametersGenerator;
+import org.bouncycastle.crypto.generators.PKCS5S2ParametersGenerator;
+import org.bouncycastle.crypto.params.KeyParameter;
+import org.bouncycastle.util.encoders.Base64;
+
+import java.security.SecureRandom;
+import java.util.regex.Pattern;
+
+public class Password {
+
+  // Min 8 characters, at least one uppercase, one lowercase, one number
+  public static Pattern PASSWORD_PATTERN = Pattern.compile("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)[a-zA-Z\\d]{8,}$");
```

Comment:
This is okay for now, but this should be handled by the AuthService which has a method for validating users, including their passwords. Then we can make this configurable. I'll show/explain to you how XHome did this.

Reply:
Thanks